api/src/main/java/com/epam/pipeline/manager/pipeline/documents/templates/processors/versionedstorage/processor/VSReportTemplateDiffProcessor.java

```diff
@@ -0,0 +1,313 @@
+        }
+        paragraph.createRun().addBreak(BreakType.TEXT_WRAPPING);
+    }
+
+    private XWPFParagraph addDescription(final XWPFParagraph paragraph, final String fontFamily, final int fontSize,
+                                         final CommitDiffsGrouping.GroupType type, final GitDiffEntry diffEntry) {
+        if (type.equals(CommitDiffsGrouping.GroupType.BY_COMMIT)) {
+            final String file = diffEntry.getDiff().getFromFileName().contains("/dev/null")
+                    ? diffEntry.getDiff().getToFileName()
+                    : diffEntry.getDiff().getFromFileName();
+            insertTextData(paragraph, file, true, fontFamily, fontSize, false);
+            insertTextData(paragraph, " file changes:", false, fontFamily, fontSize, true);
+        } else {
+            insertTextData(paragraph, "Changes in revision: ", false, fontFamily, fontSize, false);
+            insertTextData(
+                    paragraph,
+                    diffEntry.getCommit().getCommit().substring(0, 9),
+                    true, fontFamily, fontSize,
+                    false);
+
+            insertTextData(paragraph, " at ", false, fontFamily, fontSize, false);
+
+            insertTextData(paragraph,
+                    ReportDataExtractor.DATE_FORMAT.format(diffEntry.getCommit().getAuthorDate()),
+                    true, fontFamily, fontSize, true
+            );
+        }
+        for (String headerLine : diffEntry.getDiff().getHeaderLines()) {
+            if (StringUtils.isNotBlank(headerLine)) {
+                insertTextData(paragraph, headerLine, false, fontFamily, fontSize, true);
+            }
+        }
+
+        return generateDiffTable(paragraph, fontFamily, fontSize, diffEntry);
+    }
+
+    private XWPFParagraph generateDiffTable(final XWPFParagraph paragraph, final String fontFamily,
+                                            int fontSize, final GitDiffEntry diffEntry) {
+        if (diffEntry.getDiff().getHunks().isEmpty()) {
+            return createNewParagraph(paragraph, paragraph.getCTP().newCursor());
+        }
+
+        final XWPFTable xwpfTable = createTable(paragraph);
+
+        for (final Hunk hunk : diffEntry.getDiff().getHunks()) {
+            fillTableWithDiffHunk(fontFamily, fontSize, xwpfTable, hunk);
+        }
+        return createNewParagraph(paragraph, xwpfTable.getCTTbl().newCursor());
+    }
+
+    private void insertTextData(final XWPFParagraph paragraph, final String text, final boolean bold,
+                                final String fontFamily, final int fontSize, final boolean lineBreak) {
+        final XWPFRun run = paragraph.createRun();
+        run.setFontFamily(fontFamily);
+        run.setFontSize(fontSize);
+        run.setBold(bold);
+        run.setText(text);
+        if (lineBreak) {
+            run.addBreak(BreakType.TEXT_WRAPPING);
+        }
+    }
+
+    private void fillTableWithDiffHunk(final String fontFamily, final int fontSize,
```

Comment:
It's pretty hard to propose anything here but it sure looks like some private methods can be extracted to lighten up the code flow a bit.